### PR TITLE
Bookstore: connect to API

### PR DESCRIPTION
**Pull request by brainskev**
In this milestone I achieved the following;

- Made use of the [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) as my backend.
- Used [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) to make API calls.
- Fetch my books from Bookstore API on when the component has rendered with useEffect.
- Refactored my add book and remove book features to persist my changes in the server and modified my reducer logic to include HTTP requests to the Bookstore API

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -57,4 +59,4 @@
     "stylelint-csstree-validator": "^1.9.0",
     "stylelint-scss": "^3.21.0"
   }
-}
+}
```
Comment (BenMukebo) [inline package.json:62]: - [ ] Great job so far howver you should use the  `createAsyncThunk` in the project for creating thunk action creators. Kindly add this. Your code might looks like:

```js

import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios'; 


export const getBooks = createAsyncThunk(
  'books/getBooks',
  async () => {
    const response = await axios.get(
      'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/',
    );
    return response.data;
  },
); 
```

File: package.json [hunk 2]
```diff
@@ -3,16 +3,18 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
-    "@reduxjs/toolkit": "^1.9.2",
+    "@reduxjs/toolkit": "^1.9.0",
     "@testing-library/jest-dom": "^5.16.5",
     "@testing-library/react": "^13.4.0",
```
Comment (BenMukebo) [inline package.json:8]: - [ ] 🤔  This project involves connecting an external API to a React-Redux application. You need to connect your React-Redux application to an existing API to create and remove books in a remote server by using Redux Toolkit. Kindly address this and make sure All [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) calls are done with `fetch` OR `Axios` .All [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) calls are done with [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) OR [axios](https://axios-http.com/docs/intro) . 💯 👍🏻

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @brainskev,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻
- Gitflow was correctly used.  ✔️   
- Descriptive README file.  ✔️ 
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: package.json (hunk 1 above)
Comment (brainskev) [reply package.json:62]: @BenMukebo  In my books.js I have used the createAsyncThunk. Do you suggest that I have them configured wrongly? kindly elaborate

File: package.json (hunk 2 above)
Comment (brainskev) [reply package.json:8]: Hey  @BenMukebo  In books.js I used the Axios library for handling HTTP requests to an existing  API for my three action creators i.e addBook, removeBook, and getBooks. The state of the books is managed by the bookReducer, which updates the state based on the type of the dispatched action.

Comment (Owusu-Desmond) [inline package.json:8]: - [ ] As requested by the previous reviewer, This project involves connecting an external API to a React-Redux application. You need to connect your React-Redux application to an existing API to create and remove books in a remote server by using Redux Toolkit. Kindly address this and make sure All [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) calls are done with `fetch` OR `Axios` .All [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) calls are done with [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) OR [axios](https://axios-http.com/docs/intro) . :100: 👍🏻

File: package.json [hunk 3]
```diff
@@ -3,16 +3,18 @@
...
+    "axios": "^1.3.2",
     "prop-types": "^15.8.1",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
     "react-redux": "^8.0.5",
     "react-router-dom": "^6.4.3",
     "react-scripts": "5.0.1",
+    "redux-thunk": "^2.4.2",
     "uuid": "^9.0.0",
     "web-vitals": "^2.1.4"
```
Comment (Owusu-Desmond) [inline package.json:19]: - [ ] As requested by the previous code reviewer, you need to use the `createAsyncThunk` in the project for creating thunk action creators. Kindly add this. Your code might look like this:

```js

import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios'; 


export const getBooks = createAsyncThunk(
  'books/getBooks',
  async () => {
    const response = await axios.get(
      'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/',
    );
    return response.data;
  },
); 
```

*You can visit how to use `createAsyncThunk` in this[ blog post](https://dev.to/ifeanyichima/what-is-createasyncthunk-in-redux--mhe)*

File: package.json (hunk 1 above)
Comment (BenMukebo) [reply package.json:62]: Hey @brainskev, I could not see it. let's me check again!!

File: package.json (hunk 2 above)
Comment (BenMukebo) [reply package.json:8]: - [ ] Same thing here, 🤔 I have just check once again but I couldn't see it  !!!

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @brainskev , :wave: 

You have not implemented the changes requested by the previous code reviewer , but there are still some changes you still need to work on before you move on to the next project but you are almost there.



 ## Highlights :+1: 
:heavy_check_mark: Good commit history.
:heavy_check_mark: No linter errors.
:heavy_check_mark: Your project is well documented


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ismailco — COMMENTED** [review]
### Hi @brainskev, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**